src/main/scala/subsystem/BaseSubsystem.scala

```diff
@@ -33,8 +33,10 @@ abstract class BareSubsystemModuleImp[+L <: BareSubsystem](_outer: L) extends La
   ElaborationArtefacts.add("plusArgs", PlusArgArtefacts.serialize_cHeader)
   println(outer.dts)
 
-  def addOMArtefacts(): Unit = {
-    ElaborationArtefacts.add("objectModel.json", outer.objectModelJson)
+  def addOMArtefacts(components: Seq[OMComponent]): Unit = {
```

Comment:
I'm not understanding this API. Why are you passing a Seq of OMComponents into this method, then getting OMComponents that the BareSubsystem knows about, and finally creating an ElaborationArtefact? The part that confuses me the most is that you expect to grab a collection of OMComponents from elsewhere, so why is the BareSubsystem so special that it gets to both grab its own OMComponents and also create the ElaborationArtefact but the other places that have OMComponents aren't that special?

What if you just have the BareSubsystem be in charge of grabbing and returning its OMComponents (or registering its OMComponents with the DomCollector) and then have some other module be responsible for emitting the ElaborationArtefact?

Reply:
The L2$ cache is created with a mixin above the EnterpriseBasSubSystem.
Lets talk in person.

Reply:
changed the API